Execution build-native of goal org.graalvm.buildtools:native-maven-plugin:1.1.6:compile-no-fork failed

**Describe the bug**
After release of native-maven-plugin:1.1.6 native build starts to fail due to missing class (see logs below).


**Logs**
```
Execution build-native of goal org.graalvm.buildtools:native-maven-plugin:1.1.6:compile-no-fork failed: A required class was missing while executing org.graalvm.buildtools:native-maven-plugin:1.1.6:compile-no-fork: org/apache/maven/shared/utils/logging/MessageUtils
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.graalvm.buildtools:native-maven-plugin:1.1.6
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/A103522755/.m2/repository/org/graalvm/buildtools/native-maven-plugin/1.1.6/native-maven-plugin-1.1.6.jar
[ERROR] urls[1] = file:/Users/A103522755/.m2/repository/org/graalvm/buildtools/utils/1.1.6/utils-1.1.6.jar
[ERROR] urls[2] = file:/Users/A103522755/.m2/repository/com/github/openjson/openjson/1.0.13/openjson-1.0.13.jar
[ERROR] urls[3] = file:/Users/A103522755/.m2/repository/org/graalvm/buildtools/graalvm-reachability-metadata/1.1.6/graalvm-reachability-metadata-1.1.6.jar
[ERROR] urls[4] = file:/Users/A103522755/.m2/repository/org/cyclonedx/cyclonedx-maven-plugin/2.9.1/cyclonedx-maven-plugin-2.9.1.jar
[ERROR] urls[5] = file:/Users/A103522755/.m2/repository/org/cyclonedx/cyclonedx-core-java/9.0.5/cyclonedx-core-java-9.0.5.jar
[ERROR] urls[6] = file:/Users/A103522755/.m2/repository/commons-io/commons-io/2.16.1/commons-io-2.16.1.jar
[ERROR] urls[7] = file:/Users/A103522755/.m2/repository/org/apache/commons/commons-collections4/4.4/commons-collections4-4.4.jar
[ERROR] urls[8] = file:/Users/A103522755/.m2/repository/com/github/package-url/packageurl-java/1.5.0/packageurl-java-1.5.0.jar
[ERROR] urls[9] = file:/Users/A103522755/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-xml/2.17.2/jackson-dataformat-xml-2.17.2.jar
[ERROR] urls[10] = file:/Users/A103522755/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.17.2/jackson-core-2.17.2.jar
[ERROR] urls[11] = file:/Users/A103522755/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.17.2/jackson-annotations-2.17.2.jar
[ERROR] urls[12] = file:/Users/A103522755/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.17.2/jackson-databind-2.17.2.jar
[ERROR] urls[13] = file:/Users/A103522755/.m2/repository/org/codehaus/woodstox/stax2-api/4.2.2/stax2-api-4.2.2.jar
[ERROR] urls[14] = file:/Users/A103522755/.m2/repository/com/fasterxml/woodstox/woodstox-core/6.7.0/woodstox-core-6.7.0.jar
[ERROR] urls[15] = file:/Users/A103522755/.m2/repository/com/networknt/json-schema-validator/1.5.1/json-schema-validator-1.5.1.jar
[ERROR] urls[16] = file:/Users/A103522755/.m2/repository/com/ethlo/time/itu/1.10.2/itu-1.10.2.jar
[ERROR] urls[17] = file:/Users/A103522755/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.17.1/jackson-dataformat-yaml-2.17.1.jar
[ERROR] urls[18] = file:/Users/A103522755/.m2/repository/org/yaml/snakeyaml/2.2/snakeyaml-2.2.jar
[ERROR] urls[19] = file:/Users/A103522755/.m2/repository/commons-codec/commons-codec/1.17.1/commons-codec-1.17.1.jar
[ERROR] urls[20] = file:/Users/A103522755/.m2/repository/org/apache/commons/commons-lang3/3.17.0/commons-lang3-3.17.0.jar
[ERROR] urls[21] = file:/Users/A103522755/.m2/repository/org/apache/maven/shared/maven-dependency-tree/3.3.0/maven-dependency-tree-3.3.0.jar
[ERROR] urls[22] = file:/Users/A103522755/.m2/repository/org/apache/maven/shared/maven-dependency-analyzer/1.14.1/maven-dependency-analyzer-1.14.1.jar
[ERROR] urls[23] = file:/Users/A103522755/.m2/repository/org/ow2/asm/asm/9.7/asm-9.7.jar
[ERROR] urls[24] = file:/Users/A103522755/.m2/repository/org/twdata/maven/mojo-executor/2.4.1/mojo-executor-2.4.1.jar
[ERROR] urls[25] = file:/Users/A103522755/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
```


**System Info:**
 - OS: MacOS
 - GraalVM Version: 25.0.2-graalce
 - Java Version: 25
 - Plugin version `native-maven-plugin:1.1.6`

--- a/native-maven-plugin/docs/functional/native-image-builds.md
+++ b/native-maven-plugin/docs/functional/native-image-builds.md
@@ -69,7 +69,10 @@ mvn -Pnative native:list-libraries-missing-metadata
 
 ## 9. Console colors
 
-The native-image invocation must follow Maven's detected console color state. It must use
+When Maven's `style.color` property explicitly selects a colored or colorless mode, or when Maven
+runs non-interactively, the native-image invocation must follow that selection. It must use
 `--color=always` or `--color=never` on JDK 21 and later, and `-H:+BuildOutputColorful` or
-`-H:-BuildOutputColorful` on older versions. Explicit user build arguments come later and may
-override this detected default, adapting [§root/FS-native-builds.2](../../../docs/spec/functional/native-image-builds.md#2-command-line-construction).
+`-H:-BuildOutputColorful` on older versions. When the Maven plugin API does not expose a resolved
+console mode, the invocation must omit a color argument and let Native Image detect its output
+mode. Explicit user build arguments come later and may override Maven's exposed mode, adapting
+[§root/FS-native-builds.2](../../../docs/spec/functional/native-image-builds.md#2-command-line-construction).

--- a/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeImageMojo.java
+++ b/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeImageMojo.java
@@ -51,7 +51,6 @@ import org.apache.maven.plugins.annotations.Component;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.toolchain.ToolchainManager;
-import org.apache.maven.shared.utils.logging.MessageUtils;
 import org.codehaus.plexus.logging.Logger;
 import org.graalvm.buildtools.maven.config.ExcludeConfigConfiguration;
 import org.graalvm.buildtools.model.resources.NativeImageFlags;
@@ -300,18 +299,49 @@ public abstract class AbstractNativeImageMojo extends AbstractNativeMojo {
         return Collections.unmodifiableList(actualCliArgs);
     }
 
-    // Maven's detected console mode controls version-appropriate Native Image color flags. §FS-native-builds.9.
+    // Maven-exposed console modes control version-appropriate Native Image color flags. §FS-native-builds.9.
     private void addConsoleColorArgument(List<String> cliArgs) throws MojoExecutionException {
-        boolean colorEnabled = isColorEnabled();
+        Optional<Boolean> colorEnabled = getMavenColorSetting();
+        if (colorEnabled.isEmpty()) {
+            return;
+        }
         if (getNativeImageMajorVersion() >= 21) {
-            cliArgs.add(NativeImageFlags.COLOR + (colorEnabled ? "=always" : "=never"));
+            cliArgs.add(NativeImageFlags.COLOR + (colorEnabled.get() ? "=always" : "=never"));
         } else {
-            cliArgs.add(colorEnabled ? NativeImageFlags.BUILD_OUTPUT_COLORFUL : NativeImageFlags.BUILD_OUTPUT_COLORLESS);
+            cliArgs.add(colorEnabled.get()
+                    ? NativeImageFlags.BUILD_OUTPUT_COLORFUL
+                    : NativeImageFlags.BUILD_OUTPUT_COLORLESS);
         }
     }
 
-    protected boolean isColorEnabled() {
-        return MessageUtils.isColorEnabled();
+    protected Optional<Boolean> getMavenColorSetting() {
+        String styleColor = session.getUserProperties().getProperty("style.color");
+        if (styleColor == null) {
+            styleColor = session.getSystemProperties().getProperty("style.color");
+        }
+        Optional<Boolean> explicitStyleColor = parseMavenColorMode(styleColor);
+        if (explicitStyleColor.isPresent()) {
+            return explicitStyleColor;
+        }
+
+        if (!session.getRequest().isInteractiveMode()) {
+            return Optional.of(false);
+        }
+        return Optional.empty();
+    }
+
+    private static Optional<Boolean> parseMavenColorMode(String colorMode) {
+        if ("always".equalsIgnoreCase(colorMode)
+                || "yes".equalsIgnoreCase(colorMode)
+                || "force".equalsIgnoreCase(colorMode)) {
+            return Optional.of(true);
+        }
+        if ("never".equalsIgnoreCase(colorMode)
+                || "no".equalsIgnoreCase(colorMode)
+                || "none".equalsIgnoreCase(colorMode)) {
+            return Optional.of(false);
+        }
+        return Optional.empty();
     }
 
     protected int getNativeImageMajorVersion() throws MojoExecutionException {

--- a/native-maven-plugin/src/test/groovy/org/graalvm/buildtools/maven/AbstractNativeImageMojoTest.groovy
+++ b/native-maven-plugin/src/test/groovy/org/graalvm/buildtools/maven/AbstractNativeImageMojoTest.groovy
@@ -1,5 +1,8 @@
 package org.graalvm.buildtools.maven
 
+import org.apache.maven.execution.DefaultMavenExecutionRequest
+import org.apache.maven.execution.DefaultMavenExecutionResult
+import org.apache.maven.execution.MavenSession
 import org.apache.maven.plugin.MojoExecutionException
 import org.graalvm.buildtools.model.resources.NativeImageFlags
 import spock.lang.Issue
@@ -37,13 +40,14 @@ class AbstractNativeImageMojoTest extends Specification {
         ]
     }
 
-    // Maven console mode selects the Native Image version's color argument. §FS-native-builds.9.
-    @Issue("https://github.com/graalvm/native-build-tools/issues/366")
-    def "uses Maven's #label console color mode with JDK #nativeImageMajorVersion"() {
+    // An explicit Maven style selects the Native Image version's color argument. §FS-native-builds.9.
+    @Issue(["https://github.com/graalvm/native-build-tools/issues/366",
+            "https://github.com/graalvm/native-build-tools/issues/1000"])
+    def "uses Maven's explicit #styleColor style with JDK #nativeImageMajorVersion"() {
         given:
         def mojo = newMojo([])
         mojo.imageClasspath.add(testDirectory.resolve("application.jar"))
-        mojo.colorEnabled = colorEnabled
+        mojo.session.userProperties.setProperty("style.color", styleColor)
         mojo.nativeImageMajorVersion = nativeImageMajorVersion
 
         when:
@@ -53,20 +57,58 @@ class AbstractNativeImageMojoTest extends Specification {
         args.contains(expectedColorArgument)
 
         where:
-        label      | colorEnabled | nativeImageMajorVersion | expectedColorArgument
-        "disabled" | false        | 17                      | NativeImageFlags.BUILD_OUTPUT_COLORLESS
-        "disabled" | false        | 21                      | "--color=never"
-        "enabled"  | true         | 17                      | NativeImageFlags.BUILD_OUTPUT_COLORFUL
-        "enabled"  | true         | 21                      | "--color=always"
+        styleColor | nativeImageMajorVersion | expectedColorArgument
+        "never"    | 17                      | NativeImageFlags.BUILD_OUTPUT_COLORLESS
+        "never"    | 21                      | "--color=never"
+        "always"   | 17                      | NativeImageFlags.BUILD_OUTPUT_COLORFUL
+        "always"   | 21                      | "--color=always"
+        "none"     | 21                      | "--color=never"
+        "force"    | 21                      | "--color=always"
     }
 
-    // Explicit build arguments retain precedence over Maven's detected color mode. §FS-native-builds.9.
+    // Maven batch mode disables Native Image colors even when no color property is present. §FS-native-builds.9.
+    def "disables colors for Maven batch mode"() {
+        given:
+        def mojo = newMojo([])
+        mojo.imageClasspath.add(testDirectory.resolve("application.jar"))
+        mojo.session.request.interactiveMode = false
+        mojo.nativeImageMajorVersion = 21
+
+        expect:
+        mojo.getBuildArgs().contains("--color=never")
+    }
+
+    // Unresolved color selection remains Native Image's responsibility without Maven implementation classes. §FS-native-builds.9.
+    @Issue("https://github.com/graalvm/native-build-tools/issues/1000")
+    def "omits a color argument when Maven color mode is #styleColor"() {
+        given:
+        def mojo = newMojo([])
+        mojo.imageClasspath.add(testDirectory.resolve("application.jar"))
+        if (styleColor != null) {
+            mojo.session.userProperties.setProperty("style.color", styleColor)
+        }
+
+        when:
+        def args = mojo.getBuildArgs()
+
+        then:
+        !args.any {
+            it.startsWith(NativeImageFlags.COLOR) ||
+                    it == NativeImageFlags.BUILD_OUTPUT_COLORFUL ||
+                    it == NativeImageFlags.BUILD_OUTPUT_COLORLESS
+        }
+
+        where:
+        styleColor << [null, "auto"]
+    }
+
+    // Explicit build arguments retain precedence over Maven's explicit color mode. §FS-native-builds.9.
     @Issue("https://github.com/graalvm/native-build-tools/issues/366")
-    def "places explicit color build arguments after Maven's detected default"() {
+    def "places explicit color build arguments after Maven's explicit selection"() {
         given:
         def mojo = newMojo(["--color=never"])
         mojo.imageClasspath.add(testDirectory.resolve("application.jar"))
-        mojo.colorEnabled = true
+        mojo.session.userProperties.setProperty("style.color", "always")
         mojo.nativeImageMajorVersion = 21
 
         when:
@@ -114,11 +156,17 @@ class AbstractNativeImageMojoTest extends Specification {
         mojo.buildArgs = buildArgs
         mojo.configFiles = []
         mojo.useArgFile = false
+        def userProperties = new Properties()
+        def systemProperties = new Properties()
+        def request = new DefaultMavenExecutionRequest()
+                .setUserProperties(userProperties)
+                .setSystemProperties(systemProperties)
+                .setInteractiveMode(true)
+        mojo.session = new MavenSession(null, null, request, new DefaultMavenExecutionResult())
         mojo
     }
 
     private static class TestNativeImageMojo extends AbstractNativeImageMojo {
-        boolean colorEnabled
         int nativeImageMajorVersion = 25
 
         @Override
@@ -132,11 +180,6 @@ class AbstractNativeImageMojoTest extends Specification {
 
         @Override
         protected void populateClasspath() {
-        }
-
-        @Override
-        protected boolean isColorEnabled() {
-            colorEnabled
         }
 
         @Override

